### PR TITLE
Add fields for calculating taxes from source country and option for no VAT country

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it extends the functionality of the VatlayerPlugin.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1281 As a seller located in EU I want to configure whether to use the source or destination tax rates for each country](https://app.clickup.com/t/2549495/SALEOR-1281)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/plugins/vatlayer/plugin.py [hunk 1]
```diff
@@ -48,8 +49,39 @@ class VatlayerPlugin(BasePlugin):
...
+
+    DEFAULT_CONFIGURATION = [
+        {"name": "Access key", "value": None},
+        {"name": "source_country", "value": None},
+        {"name": "countries_to_calculate_taxes_from_source", "value": None},
+        {"name": "excluded_countries", "value": None},
+    ]
+
     CONFIG_STRUCTURE = {
+        "source_country": {
```
Comment (maarcingebala) [inline saleor/plugins/vatlayer/plugin.py:61]: I think in the VAT-terminology, this is called "origin country" ([source](https://www.europarl.europa.eu/RegData/etudes/BRIE/2017/603265/EPRS_BRI(2017)603265_EN.pdf), see the "Summary" section). I would rename it here and in the labels and help texts.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (fowczarek) [inline CHANGELOG.md:1]: :policeman:

File: saleor/plugins/vatlayer/plugin.py [hunk 3]
```diff
@@ -48,8 +49,39 @@ class VatlayerPlugin(BasePlugin):
...
+            "type": ConfigurationTypeField.STRING,
+            "help_test": (
+                "Country code in ISO format, required to calculate taxes for countries "
+                "from `Calculate taxes based on source country`."
```
Comment (maarcingebala) [inline saleor/plugins/vatlayer/plugin.py:65]: `Calculate taxes based on source country` - there is no such label.

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/vatlayer/plugin.py [hunk 4]
```diff
@@ -118,10 +174,20 @@ def _get_taxes_for_country(self, country: Country):
         from cache or db.
         """
         if not country:
-            country = Country(settings.DEFAULT_COUNTRY)
+            source_country_code = self.config.source_country or settings.DEFAULT_COUNTRY
```
Comment (maarcingebala) [inline saleor/plugins/vatlayer/plugin.py:177]: There is also the `company_address` defined in the SiteSettings model, maybe we want to use it here as well?

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**